api/models/spell.py

```diff
@@ -0,0 +1,229 @@
+        spell_level_name_lookup = [
+            'Cantrip',
+            '1st-level',
+            '2nd-level',
+            '3rd-level',
+            '4th-level',
+            '5th-level',
+            '6th-level',
+            '7th-level',
+            '8th-level',
+            '9th-level'
+        ]
+        return spell_level_name_lookup[self.spell_level]
+
+    # Stored data about casting the spell
+    casting_time = models.TextField(
+        help_text='Amount of time it takes to cast the spell, such as "1 bonus action" or "4 hours".')
+
+    range = models.TextField(help_text='Text description of the target range.')
+
+    target_range_sort = models.IntegerField(
+        validators=[MinValueValidator(0)],
+        help_text='Sortable distance ranking to the target.')
+
+    requires_verbal_components = models.BooleanField(
+        help_text='Casting this spell requires verbal components.')
+    requires_somatic_components = models.BooleanField(
+        help_text='Casting this spell requires somatic components.')
+    requires_material_components = models.BooleanField(
+        help_text='Casting this spell requires material components.')
+
+    material = models.TextField(
+        help_text='Description of the material required.')
+
+    higher_level = models.TextField(
+        help_text='What happens if you cast this at a higher level.')
+
+    can_be_cast_as_ritual = models.BooleanField(
+        help_text='Whether or not the spell can be cast as a ritual.')
+
+    # Computed data about casting the spell
+    def v1_components(self):
+        """Presents the components in a standard list like "V, S, M."""
+        components_list = []
+        if self.requires_verbal_components:
+            components_list.append("V")
+        if self.requires_somatic_components:
+            components_list.append("S")
+        if self.requires_material_components:
+            components_list.append("M")
+
+        return ', '.join(components_list)
+
+    def v1_ritual(self):
+        """Field calculated and presented to maintain compatibility with original API."""
+        if self.can_be_cast_as_ritual:
+            return "yes"
+        else:
+            return "no"
+
+    # Stored data about the spell's effect
+    duration = models.TextField(
+        help_text='Description of the duration such as "instantaneous" or "Up to 1 minute"')
+
+    requires_concentration = models.BooleanField(
+        help_text='Whether the spell requires concentration')
+
+    def v1_concentration(self):
+        """Field calculated and presented to maintain compatibility with original API."""
+        if self.requires_concentration:
+            return "yes"
+        else:
+            return "no"
+
+    archetype = models.TextField(
+        help_text='Archetype that can learn this spell. If empty, assume all archetypes.')
+    circles = models.TextField(
+        help_text='Druid Archetypes that can learn this spell.')
```

Comment:
Why is this separate from Archetypes? Circles are a subset of Archetypes.

Reply:
Yeah that needs resolution. Not in scope for this PR though.